File: benchmarks/go/persistence.go
package main

import (
	"encoding/json"
	"os"
)

func SaveTasks(tasks map[uint64]*Task, filepath string) error {
	data, err := json.MarshalIndent(tasks, "", "  ")
	if err != nil {
		return err
	}

	return os.WriteFile(filepath, data, 0644)
}

func LoadTasks(filepath string) (map[uint64]*Task, error) {
	data, err := os.ReadFile(filepath)
	if err != nil {
		return nil, err
	}

	var tasks map[uint64]*Task
	if err := json.Unmarshal(data, &tasks); err != nil {
		return nil, err
	}

	return tasks, nil
}

File: benchmarks/rust/src/persistence.rs
use std::collections::HashMap;
use std::fs;
use std::io;
use std::path::Path;

use crate::models::Task;

pub fn save_tasks(path: &Path, tasks: &HashMap<u64, Task>) -> io::Result<()> {
    let json = serde_json::to_string_pretty(tasks)
        .map_err(|e| io::Error::new(io::ErrorKind::InvalidData, e))?;
    fs::write(path, json)
}

pub fn load_tasks(path: &Path) -> io::Result<HashMap<u64, Task>> {
    if !path.exists() {
        return Ok(HashMap::new());
    }
    let data = fs::read_to_string(path)?;
    let tasks: HashMap<u64, Task> = serde_json::from_str(&data)
        .map_err(|e| io::Error::new(io::ErrorKind::InvalidData, e))?;
    Ok(tasks)
}
